[Design] [SD] Scope (#424)
arkinmodi/project-sayyara: docs/Design/SystDesign/SystDes.tex

\begin{tabular}{l l}
	\toprule
	\textbf{symbol} & \textbf{description}                \\
	\midrule
	\progname       & Explanation of program name         \\
	MIS             & Module Interface Specifications     \\
	MG              & Module Guide                        \\

> 	SRS             & Software Requirements Specification \\

Reviewer: Add PWA

\section{Scope}

> The system is designed to connect vehicle owners to independent shop owners, providing users with

Reviewer: ```suggestion
The system is designed to connect vehicle owners and independent shop owners, providing users with
```

> the ability to communicate with one another and create, schedule and see pricing for vehicle

Reviewer: ```suggestion
the ability to communicate with one another, and respectively view and manage the interactions and processes involved in a typical auto repair and maintenance service experience.  
```
We can probably keep this brief and high-level since it is expanded upon in normal behaviour. 

We can probably removed the canned/custom part?

Author: Yeah I guess so, didn't realize it was being expanded on elsewhere

> processes involved in a typical auto repair and maintenance service experience. services, which can

Reviewer: ```suggestion
processes involved in a typical auto repair and maintenance service experience.
```

> services, which can either be canned services or custom services. All functionality of the system

Reviewer: Clean up and formatting

> either be canned services or custom services. All functionality of the system has been defined in

Reviewer: ```suggestion
All functionality of the system has been defined in
```

\url{https://github.com/arkinmodi/project-sayyara/blob/main/docs/SRS/SRS.pdf} and everything not
included in the SRS is not part of the scope.

The system includes a PWA and the relevant database to store information relevant to the
application.

\subsection{Context Diagram}

Below is a context diagram detailing the boundary between the system and the environment around it.

\begin{figure}[H]
	\centering
	\includegraphics[width=\textwidth]{./diagrams/ContextDiagram.png}

> 	\caption{Context Diagram}

Reviewer: Should front-end and backend have arrows pointing to each other?

Author: I thought it wasn't needed there because they were both encapsulated in the Sayyara application, but if you think it's necessary I can add it

Reviewer: @arkinmodi thoughts?

Author: I can change it anyways, it might be better to explicitly state that, it's just that the context diagram is more concerned with the environment around the system

Reviewer: I leave it up to you and @arkinmodi. Otherwise LGTM

Reviewer 2: I think the arrow of the backend pointing to the frontend might imply that the backend can call the frontend. I would remove it.

> 	\label{ContextDiagram}

Reviewer: One more thing, should Vercel server be pointing to the PWA as a whole and not back-end? It hosts both right? Is the DB also hosted on Vercel @arkinmodi? If so, maybe it can just point to Sayyara System

Author: unsure, I'll let @arkinmodi make that decision, not too familiar with Vercel

Reviewer 2: I think to make it simple, just point to Sayyara PWA as a whole.

Reviewer 2: The database is hosted on https://planetscale.com/ which is a different company.
